Challenges/GoDownTheRabbitHole/Solutions/RioMH.js

```diff
@@ -0,0 +1,59 @@
+let parentsCache = [];
+
+function countParents(parentId, data) {
+    let cached = parentsCache[parentId];
+    let count = 1;
+
+    if (cached) {
+        count = cached;
+    } else {
+        for (let index = 1; index < data.length; index++) {
+            let item = data[index]
+            if (item.itemId == parentId && item.parentId != null) {
+                count += countParents(item.parentId, data);
+                break;
+            }
+        }
+    }
+
+    parentsCache[parentId] = count;
+    return count;
+}
+
+function getIndent(item, data) {
+    let indent = "";
+
+    if (item.parentId != 0) {
+        let repeatCount = countParents(item.parentId, data);
```

Comment:
Could avoid a function call for cached results here.
`let repeatCount = parentsCache[item.parentId] || countParents(item.parentId, data)`
Might by hacky and need a comment to describe what's going on.

Reply:
Seems fine to me, doesn't feel hacky at all ;) The variables are verbose enough for it to be understood without comments too IMO. It'll clean up the `countParents` function a lot (although it will mess with the organisation a bit).